spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/TelemetryUtils.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.autoconfigure.telemetry;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+
+import static com.microsoft.applicationinsights.core.dependencies.apachecommons.codec.digest.DigestUtils.sha256Hex;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TelemetryUtils {
+
+    private static final String UNKNOWN_MAC = "Unknown-Mac-Address";
+
+    private static String getMacAddress() {
+        try {
+            final InetAddress ip = InetAddress.getLocalHost();
```

Comment:
Sorry that I should have brought this up early. 
1.  Is localhost always ip? Could it be a hostname? If this, rename ip into localhost.
2. `network` is also confusing. Maybe we can chain method call without introducing more variable and complexity.